[GTK] Generate serialization of GtkSettingsState

csaavedra opened the pull request:
#### 1c24b9632e2a589a79d37ec8c772bb10e91847d6
<pre>
[GTK] Generate serialization of GtkSettingsState
<a href="https://bugs.webkit.org/show_bug.cgi?id=272982">https://bugs.webkit.org/show_bug.cgi?id=272982</a>

Reviewed by Michael Catanzaro.

* Source/WebKit/Shared/gtk/ArgumentCodersGtk.serialization.in:
* Source/WebKit/Shared/gtk/GtkSettingsState.cpp: Removed.
* Source/WebKit/Shared/gtk/GtkSettingsState.h:
* Source/WebKit/SourcesGTK.txt:

Canonical link: <a href="https://commits.webkit.org/277743@main">https://commits.webkit.org/277743@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b5b19c5c78a7b7050adc169764b15f3eda4aabd8

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/48450 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/27662 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/51409 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/51138 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/44515 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/33598 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/25184 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/39606 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/49032 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/25341 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/41825 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/20728 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/22824 "Passed tests") | | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/6507 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/44766 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/43477 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/53043 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/23497 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/19823 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/46922 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/24762 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/42023 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/45842 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/25567 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/6896 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/24485 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

GeorgesStavracas commented on Source/WebKit/Shared/gtk/ArgumentCodersGtk.serialization.in:
```diff
@@ -30,3 +30,18 @@ class WebCore::SelectionData {
+    std::optional<int> cursorBlinkTime;
+    std::optional<bool> primaryButtonWarpsSlider;
+    std::optional<bool> overlayScrolling;
+    std::optional<bool> enableAnimations;
```
`enableAnimations` doesn't seem to be in the encode nor decoded methods, but it is in the GtkSettingsState struct. Not sure if it's a problem?

csaavedra replied:
I checked `git blame` and that member was added later, by @TingPing . I wonder if it was just an overlook?

TingPing replied:
Would have been. Something these prevent :)

csaavedra committed:
[GTK] Generate serialization of GtkSettingsState
https://bugs.webkit.org/show_bug.cgi?id=272982

Reviewed by Michael Catanzaro.

* Source/WebKit/Shared/gtk/ArgumentCodersGtk.serialization.in:
* Source/WebKit/Shared/gtk/GtkSettingsState.cpp: Removed.
* Source/WebKit/Shared/gtk/GtkSettingsState.h:
* Source/WebKit/SourcesGTK.txt:

Canonical link: https://commits.webkit.org/277743@main
